src/shared/utils/pagination.utils.ts

```diff
@@ -0,0 +1,19 @@
+import { Field, ObjectType, Int } from '@nestjs/graphql';
+import { Type } from '@nestjs/common';
+import { IPaginatedType } from '../interfaces/pagination.interface';
+import { ProductItem } from 'src/inventory/entities/product-item.entity';
+
+export function Paginated<T>(classRef: Type<T>): Type<IPaginatedType<T>> {
+    @ObjectType({ isAbstract: true })
+    abstract class PaginatedType implements IPaginatedType<T> {
```

Comment:
As a rule of thumb I've discussed with @nk-coding:
Every field in the GraphQL schema needs documentation, even if it is only `the <name>`.
JSDoc comments can be omitted when there is already GraphQL documentation.